src/libraries/System.Text.Json/gen/Reflection/CustomAttributeDataWrapper.cs

```diff
@@ -13,7 +13,10 @@ internal class CustomAttributeDataWrapper : CustomAttributeData
     {
         public CustomAttributeDataWrapper(AttributeData a, MetadataLoadContextInternal metadataLoadContext)
         {
-            Debug.Assert(a.AttributeConstructor != null);
+            if (a.AttributeConstructor is null)
+            {
+                throw new InvalidOperationException();
```

Comment:
nit: might be worth to add some message

Reply:
It's repeating the same pattern as other exceptions within the source generator. Exceptions like these are handled gracefully by the IDE, unlike failing debug assertions.